src/com/intellij/idea/plugin/hybris/actions/CopyFileToHybrisConsoleUtils.kt

```diff
@@ -0,0 +1,123 @@
+import com.intellij.idea.plugin.hybris.toolwindow.CopyFileToHybrisConsoleDialog
+import com.intellij.idea.plugin.hybris.toolwindow.HybrisToolWindowFactory
+import com.intellij.idea.plugin.hybris.toolwindow.HybrisToolWindowService.Companion.getInstance
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Disposer
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiManager
+import java.util.*
+import javax.swing.tree.DefaultMutableTreeNode
+import javax.swing.tree.TreePath
+
+object CopyFileToHybrisConsoleUtils {
+
+    @JvmStatic
+    fun copySelectedFilesToHybrisConsole(project: Project, consoleTitle: String, dialogTitle: String) {
+        getInstance(project).consolesPanel.findConsole(consoleTitle)
+            ?.let { hybrisConsole ->
+                val isConsoleNotEmpty = getTextFromHybrisConsole(project, hybrisConsole).isNotEmpty()
+                val query = getQueryFromSelectedFiles(project)
+
+                val copyToHybrisConsoleFn = { copyToHybrisConsole(project, consoleTitle, query) }
+                CopyFileToHybrisConsoleDialog(
+                    project,
+                    HybrisI18NBundleUtils.message(DIALOG_TITLE + dialogTitle)
+                )
+                    .takeIf { isConsoleNotEmpty }
+                    ?.show(copyToHybrisConsoleFn) ?: copyToHybrisConsoleFn()
+            }
+    }
+
+    private fun getTextFromHybrisConsole(project: Project, hybrisConsole: HybrisConsole) =
+        ConsoleExecutionEditor(LanguageConsoleImpl.Helper(project, hybrisConsole.virtualFile))
+            .also { Disposer.dispose(it) }
+            .document.text
+
+    private fun copyToHybrisConsole(project: Project, consoleTitle: String, query: String) {
+        getInstance(project).consolesPanel.findConsole(consoleTitle)
+            ?.let { hybrisConsole ->
+                with(getInstance(project)) {
+                    activateToolWindow()
+                    activateToolWindowTab(HybrisToolWindowFactory.CONSOLES_ID)
+                    consolesPanel.setActiveConsole(hybrisConsole)
+                }
+                hybrisConsole.clear()
+                hybrisConsole.setInputText(query)
+            }
+    }
+
+    private fun getQueryFromSelectedFiles(project: Project): String {
+        return getSelectedFiles(project)
+            .mapNotNull { PsiManager.getInstance(project).findFile(it) }
+            .mapNotNull { it.text }
+            .joinToString(System.lineSeparator())
+    }
+
+    @JvmStatic
+    fun isRequiredSingleFileExtension(project: Project, fileExtension: String) =
+        getFileExtensions(project) == listOf(fileExtension)
+
+    @JvmStatic
+    fun isRequiredMultipleFileExtension(project: Project, fileExtension: String) =
+        getFileExtensions(project).all { it == fileExtension }
+
+
+    private fun getFileExtensions(project: Project): List<String?> {
+        return getSelectedFiles(project)
+            .filter { !it.isDirectory }
```

Comment:
not exactly the same logic, please compare with the original code

Reply:
invalid for current version